framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3244,8 +3245,8 @@ private void checkPreAndPostConditions() {
 
             // Check postconditions
             ContractsUtils contracts = ContractsUtils.getInstance(atypeFactory);
-            Set<Postcondition> superPost = contracts.getPostconditions(overridden.getElement());
-            Set<Postcondition> subPost = contracts.getPostconditions(overrider.getElement());
+            Set<Contract> superPost = contracts.getPostconditions(overridden.getElement());
```

Comment:
@mernst Did you get to try to improve the return types of these methods? It would be nicer if these methods and the code above didn't use the general `Contract` and casts and instead used the more precise types.
You can probably achieve this by having helper methods that have a generic return type and then the existing methods are simple wrappers around the generic method, with the precise return type.

Reply:
@wmdietl I did not do so.

Reply:
Feel free to make that change in a subsequent pull request.  Thanks!